<ui version="4.0" >
 <author>Toni Gundogdu</author>
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>543</width>
    <height>284</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>abby</string>
  </property>
  <widget class="QWidget" name="centralwidget" >
   <layout class="QGridLayout" name="gridLayout_5" >
    <item row="0" column="0" >
     <widget class="QTabWidget" name="tabWidget" >
      <property name="currentIndex" >
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab" >
       <attribute name="title" >
        <string>&amp;Video</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3" >
        <item row="0" column="0" >
         <layout class="QHBoxLayout" name="horizontalLayout" >
          <item>
           <widget class="QLabel" name="label" >
            <property name="text" >
             <string>Video &amp;Link:</string>
            </property>
            <property name="buddy" >
             <cstring>urlEdit</cstring>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="urlEdit" >
            <property name="toolTip" >
             <string>Link to the video page to download the video from.

Example: http://youtube.com/watch?v=3HD220e0bx4</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pasteButton" >
            <property name="toolTip" >
             <string>Paste</string>
            </property>
            <property name="text" >
             <string/>
            </property>
            <property name="icon" >
             <iconset resource="abby.qrc" >
              <normaloff>:/edit-paste.png</normaloff>:/edit-paste.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="startButton" >
            <property name="text" >
             <string>&amp;Start</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="1" column="0" >
         <spacer name="verticalSpacer" >
          <property name="orientation" >
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0" >
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="2" column="0" >
         <widget class="QTabWidget" name="tabWidget_2" >
          <property name="tabPosition" >
           <enum>QTabWidget::South</enum>
          </property>
          <property name="currentIndex" >
           <number>0</number>
          </property>
          <widget class="QWidget" name="tab_3" >
           <attribute name="title" >
            <string>&amp;Progress</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout" >
            <item row="0" column="0" >
             <layout class="QHBoxLayout" name="horizontalLayout_2" >
              <item>
               <widget class="QLabel" name="label_2" >
                <property name="sizePolicy" >
                 <sizepolicy vsizetype="Preferred" hsizetype="Maximum" >
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text" >
                 <string>File:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="fileLabel" >
                <property name="sizePolicy" >
                 <sizepolicy vsizetype="Preferred" hsizetype="Preferred" >
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text" >
                 <string>-</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="1" column="0" >
             <layout class="QHBoxLayout" name="horizontalLayout_3" >
              <item>
               <widget class="QLabel" name="label_3" >
                <property name="sizePolicy" >
                 <sizepolicy vsizetype="Preferred" hsizetype="Maximum" >
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text" >
                 <string>Size:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="sizeLabel" >
                <property name="enabled" >
                 <bool>true</bool>
                </property>
                <property name="text" >
                 <string>-- / --</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_4" >
                <property name="sizePolicy" >
                 <sizepolicy vsizetype="Preferred" hsizetype="Maximum" >
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text" >
                 <string>Rate:</string>
                </property>
                <property name="alignment" >
                 <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="rateLabel" >
                <property name="enabled" >
                 <bool>true</bool>
                </property>
                <property name="text" >
                 <string>--.-</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_5" >
                <property name="sizePolicy" >
                 <sizepolicy vsizetype="Preferred" hsizetype="Maximum" >
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text" >
                 <string>ETA:</string>
                </property>
                <property name="alignment" >
                 <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="etaLabel" >
                <property name="enabled" >
                 <bool>true</bool>
                </property>
                <property name="text" >
                 <string>--:--</string>
                </property>
                <property name="alignment" >
                 <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="2" column="0" >
             <layout class="QHBoxLayout" name="horizontalLayout_4" >
              <item>
               <widget class="QProgressBar" name="progressBar" >
                <property name="enabled" >
                 <bool>true</bool>
                </property>
                <property name="sizePolicy" >
                 <sizepolicy vsizetype="Maximum" hsizetype="Preferred" >
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="value" >
                 <number>0</number>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="cancelButton" >
                <property name="enabled" >
                 <bool>false</bool>
                </property>
                <property name="sizePolicy" >
                 <sizepolicy vsizetype="Maximum" hsizetype="Maximum" >
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text" >
                 <string>&amp;Cancel</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab_4" >
           <attribute name="title" >
            <string>Lo&amp;g</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_2" >
            <item row="0" column="0" >
             <widget class="QTextEdit" name="logEdit" >
              <property name="enabled" >
               <bool>true</bool>
              </property>
              <property name="readOnly" >
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2" >
       <attribute name="title" >
        <string>&amp;Options</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4" >
        <item row="0" column="0" >
         <layout class="QHBoxLayout" name="horizontalLayout_5" >
          <item>
           <widget class="QCheckBox" name="saveasBox" >
            <property name="toolTip" >
             <string>Write video to file. Overrides default file naming.</string>
            </property>
            <property name="text" >
             <string>&amp;Save video as:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="saveasEdit" >
            <property name="enabled" >
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="saveasButton" >
            <property name="enabled" >
             <bool>false</bool>
            </property>
            <property name="text" >
             <string>&amp;Browse</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="1" column="0" >
         <layout class="QHBoxLayout" name="horizontalLayout_6" >
          <item>
           <widget class="QCheckBox" name="continueBox" >
            <property name="toolTip" >
             <string>Continue a partially downloaded video file.</string>
            </property>
            <property name="text" >
             <string>&amp;Continue partially downloaded file</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_2" >
            <property name="orientation" >
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0" >
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="label_9" >
            <property name="sizePolicy" >
             <sizepolicy vsizetype="Preferred" hsizetype="Maximum" >
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text" >
             <string>V&amp;ideo Format:</string>
            </property>
            <property name="buddy" >
             <cstring>formatCombo</cstring>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="formatCombo" >
            <property name="sizePolicy" >
             <sizepolicy vsizetype="Fixed" hsizetype="Maximum" >
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip" >
             <string>Download format of the video. Note that different format
typically means different video resolution. Also note
that some formats may not always be available.</string>
            </property>
            <item>
             <property name="text" >
              <string>flv</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
        <item row="5" column="0" >
         <layout class="QHBoxLayout" name="horizontalLayout_9" >
          <item>
           <widget class="QCheckBox" name="streamBox" >
            <property name="toolTip" >
             <string>Run the stream command defined in the preferences when
the file transfer reaches the specified percent done.

Note that if the started command runs out of downloaded
video data, abby does not attempt to restore the last
state. It is recommended that you use this feature
with broadband connections only.</string>
            </property>
            <property name="text" >
             <string>S&amp;tart streaming at:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="streamSpin" >
            <property name="enabled" >
             <bool>false</bool>
            </property>
            <property name="sizePolicy" >
             <sizepolicy vsizetype="Fixed" hsizetype="Maximum" >
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip" >
             <string>Stream command is started when file transfer
reaches this percentage. Be sure to set it 
high enough to make sure that there is enough
video data for the command to process.</string>
            </property>
            <property name="suffix" >
             <string>%</string>
            </property>
            <property name="minimum" >
             <number>1</number>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_4" >
            <property name="orientation" >
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0" >
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item row="6" column="0" >
         <layout class="QHBoxLayout" name="horizontalLayout_7" >
          <item>
           <widget class="QCheckBox" name="commandBox" >
            <property name="toolTip" >
             <string>Run the command specified in the preferences after
download finishes.</string>
            </property>
            <property name="text" >
             <string>&amp;Run external command when download finishes</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer" >
            <property name="orientation" >
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0" >
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item row="7" column="0" >
         <spacer name="verticalSpacer_2" >
          <property name="orientation" >
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0" >
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="2" column="0" >
         <widget class="QCheckBox" name="titleBox" >
          <property name="toolTip" >
           <string>Use video page title for naming downloaded files.
See also the Preferences for the character-class
used to clean up the video page titles.

Note that cclive must be built WITH_PERL=yes for
this to work.</string>
          </property>
          <property name="text" >
           <string>&amp;Use video page title for naming</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar" >
   <property name="geometry" >
    <rect>
     <x>0</x>
     <y>0</y>
     <width>543</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File" >
    <property name="title" >
     <string>&amp;File</string>
    </property>
    <addaction name="action_About" />
    <addaction name="separator" />
    <addaction name="action_Quit" />
   </widget>
   <widget class="QMenu" name="menu_Edit" >
    <property name="title" >
     <string>&amp;Edit</string>
    </property>
    <addaction name="action_Preferences" />
   </widget>
   <addaction name="menu_File" />
   <addaction name="menu_Edit" />
  </widget>
  <widget class="QStatusBar" name="statusbar" />
  <action name="action_Quit" >
   <property name="text" >
    <string>&amp;Quit</string>
   </property>
   <property name="statusTip" >
    <string>Quit abby</string>
   </property>
   <property name="shortcut" >
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="action_Preferences" >
   <property name="enabled" >
    <bool>true</bool>
   </property>
   <property name="text" >
    <string>Prefere&amp;nces...</string>
   </property>
   <property name="statusTip" >
    <string>Edit preferences</string>
   </property>
   <property name="shortcut" >
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="action_About" >
   <property name="text" >
    <string>&amp;About...</string>
   </property>
   <property name="statusTip" >
    <string>About abby</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>urlEdit</tabstop>
  <tabstop>startButton</tabstop>
  <tabstop>tabWidget_2</tabstop>
  <tabstop>cancelButton</tabstop>
  <tabstop>logEdit</tabstop>
  <tabstop>saveasBox</tabstop>
  <tabstop>saveasEdit</tabstop>
  <tabstop>saveasButton</tabstop>
  <tabstop>continueBox</tabstop>
  <tabstop>formatCombo</tabstop>
  <tabstop>titleBox</tabstop>
  <tabstop>streamBox</tabstop>
  <tabstop>streamSpin</tabstop>
  <tabstop>commandBox</tabstop>
  <tabstop>tabWidget</tabstop>
 </tabstops>
 <resources>
  <include location="abby.qrc" />
 </resources>
 <connections>
  <connection>
   <sender>startButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onStart()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>521</x>
     <y>97</y>
    </hint>
    <hint type="destinationlabel" >
     <x>189</x>
     <y>148</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cancelButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onCancel()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>510</x>
     <y>180</y>
    </hint>
    <hint type="destinationlabel" >
     <x>189</x>
     <y>148</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_About</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>onAbout()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel" >
     <x>189</x>
     <y>151</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>streamBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>onStreamStateChanged(int)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>81</x>
     <y>191</y>
    </hint>
    <hint type="destinationlabel" >
     <x>253</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Quit</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel" >
     <x>189</x>
     <y>148</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveasButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onSaveasBrowse()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>521</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel" >
     <x>244</x>
     <y>160</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Preferences</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>onPreferences()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel" >
     <x>253</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>urlEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>onURLEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>252</x>
     <y>82</y>
    </hint>
    <hint type="destinationlabel" >
     <x>253</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>formatCombo</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>onFormatStateChanged(int)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>521</x>
     <y>129</y>
    </hint>
    <hint type="destinationlabel" >
     <x>253</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveasBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>onSaveasStateChanged(int)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>81</x>
     <y>82</y>
    </hint>
    <hint type="destinationlabel" >
     <x>271</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pasteButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onPasteURL()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>416</x>
     <y>84</y>
    </hint>
    <hint type="destinationlabel" >
     <x>271</x>
     <y>141</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onStart()</slot>
  <slot>onCancel()</slot>
  <slot>onAbout()</slot>
  <slot>onSaveasStateChanged(int)</slot>
  <slot>onSaveasBrowse()</slot>
  <slot>onStreamStateChanged(int)</slot>
  <slot>onPreferences()</slot>
  <slot>onURLEditingFinished()</slot>
  <slot>onFormatStateChanged(int)</slot>
  <slot>onPasteURL()</slot>
 </slots>
</ui>
